File: telerik_ui_for_winforms_courseware_chm/Courseware/RealEstate/VB/MyDocLib/CalendarPanel.vb
Imports System
Imports Telerik.WinControls.UI.Docking.RadDock

Namespace MyDockLib
    Partial Public Class CalendarPanel
        Inherits DockUserControls

        Public Sub New()
            InitializeComponent()
        End Sub

        Public Delegate Sub DateChangedHandler(ByVal [date] As DateTime)
        Public Event DateChanged As DateChangedHandler
        Private Sub radCalendar1_SelectionChanged(ByVal sender As Object, ByVal e As EventArgs)
            If DateChanged <> Nothing Then
                DateChanged((TryCast(sender, Telerik.WinControls.UI.RadCalendar)).SelectedDate)
            End If
        End Sub
    End Class
End Namespace

File: telerik_ui_for_winforms_courseware_chm/Courseware/RealEstate/CS/MyDockLib/CalendarPanel.cs
using System;

using Telerik.WinControls.RadDock;

namespace MyDockLib
{
    public partial class CalendarPanel : UserDockControl
    {
        public CalendarPanel()
        {
            InitializeComponent();
        }

        public delegate void DateChangedHandler(DateTime date);
        public event DateChangedHandler DateChanged;
        private void radCalendar1_SelectionChanged(object sender, EventArgs e)
        {
            if (DateChanged != null)
            {
                DateChanged((sender as Telerik.WinControls.UI.RadCalendar).SelectedDate);
            }
        }
    }
}
